error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/ExhaustiveRefasterTypeMigrationTest.java

```diff
@@ -0,0 +1,252 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class ExhaustiveRefasterTypeMigrationTest {
+  @Test
+  void identification() {
+    CompilationTestHelper.newInstance(ExhaustiveRefasterTypeMigration.class, getClass())
+        .addSourceLines(
+            "Util.java",
+            "class Util {",
+            "  public static int CONSTANT = 42;",
+            "",
+            "  public static void publicStaticVoidMethod() {}",
+            "",
+            "  static void packagePrivateStaticVoidMethod() {}",
+            "",
+            "  protected static void protectedStaticVoidMethod() {}",
+            "",
+            "  private static void privateStaticVoidMethod() {}",
+            "",
+            "  public static int publicStaticIntMethod2() {",
+            "    return 0;",
+            "  }",
+            "",
+            "  public String publicStringMethodWithArg(int arg) {",
+            "    return String.valueOf(arg);",
+            "  }",
+            "}")
+        .addSourceLines(
+            "A.java",
+            "import com.google.errorprone.refaster.annotation.AfterTemplate;",
+            "import com.google.errorprone.refaster.annotation.BeforeTemplate;",
+            "import tech.picnic.errorprone.refaster.annotation.TypeMigration;",
+            "",
+            "class A {",
+            "  class UnannotatedEmptyClass {}",
+            "",
+            "  // BUG: Diagnostic contains: Migration of type 'int' is unsupported",
+            "  @TypeMigration(of = int.class)",
+            "  class AnnotatedWithPrimitive {}",
+            "",
+            "  @TypeMigration(",
+            "      of = Util.class,",
+            "      unmigratedMethods = {",
+            "        \"publicStaticIntMethod2()\",",
+            "        \"publicStringMethodWithArg(int)\",",
+            "        \"publicStaticVoidMethod()\"",
+            "      })",
+            "  class AnnotatedEmptyClass {}",
+            "",
+            "  @TypeMigration(",
+            "      of = Util.class,",
+            "      unmigratedMethods = {",
+            "        \"publicStaticVoidMethod()\",",
+            "        \"publicStringMethodWithArg(int)\",",
```

Comment:
Do we need to test overloaded methods here ?
```suggestion
            "        \"publicStringMethodWithArg(int)\",",
            "        \"publicStringMethodWithArg(String)\",",
```

Reply:
Which brings me to my question on why do we need to flag all the variants of the overloaded method ? 🤔 

Is it wrong to assume that if a rule does not apply to one type set of arguments, will not be applied to other ?